[core-lro] README improvements (#8592)
Azure/azure-sdk-for-js: sdk/core/core-lro/README.md

# Azure Core LRO client library for JavaScript

> `@azure/core-lro` is a JavaScript library that allows developers to build pollers that keep track of Long Running Operations (LROs) against Azure services. Until completion, such operations require consecutive calls to Azure services to update a local representation of the remote operation status.

Reviewer: ```suggestion
`@azure/core-lro` is a JavaScript library that manages long running operations (LROs) against Azure services. Until completion, such operations require consecutive calls to Azure services to update a local representation of the remote operation status.
```

Author: Thank you!!!

> `@azure/core-lro` is a JavaScript library that allows developers to build pollers that keep track of Long Running Operations (LROs) that require consecutive calls to endpoints provided by the Azure services to update a local representation of the status of the remote operation until its completion.

Reviewer 2: Too long a sentence. Consider breaking it up for ease of readability. Below is an option, but am not sure if it ends up mis-representing what we want to say. LRO experts should review

```suggestion
`@azure/core-lro` is a JavaScript library that allows developers to build pollers that keep track of Long Running Operations (LROs) against Azure services. Until completion, such operations require consecutive calls to Azure services to update a local representation of the remote operation status.
```

Author: Thank you for reviewing this!

> **Please note:** This library is intended to be used as a dependency of official Azure SDK packages and not intended for inclusion in your projects directly.

Reviewer: ```suggestion
**Please note:** This library is included with other Azure SDK libraries that need it. It should not be used as a direct dependency in your projects.
```

Want to be more clear that users shouldn't be manually npm-installing this thing.

> Even though this library is publicly available for installation, this library is not intended to be used outside of the projects inside of the `azure-sdk-for-js` repository.

Reviewer 3: you could say "this library is intended to be used as a dependency of official Azure SDK packages and not intended for inclusion in your projects directly."

Reviewer 3: You may also want to start the sentence **Please note:**

Author: Thank you! I'll do this in a bit

`@azure/core-lro` is made following our [Long Running Operations guidelines](https://azure.github.io/azure-sdk/typescript_design.html#ts-lro)

[Source code](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/core/core-lro) | [Package (npm)](https://www.npmjs.com/package/@azure/core-lro) | [API Reference Documentation](https://docs.microsoft.com/javascript/api/@azure/core-lro) | [Samples](./samples)

## Getting started

### Install the package

> To install this library for a project under the `azure-sdk-for-js`, make sure you are at the root of that project, then use rush as follows:

Reviewer 3: link to rush

## Examples

You will be able to find some working examples of an implementation of an operation and a poller in:

- [The `@azure/core-lro` samples](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/core/core-lro/samples).
- [The `@azure/core-lro` tests](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/core/core-lro/test).

> - [Azure SDK libraries using `@azure/core-lro`](https://github.com/Azure/azure-sdk-for-js/search?q="from+%40azure%5C%2Fcore-lro"&unscoped_q="from+%40azure%5C%2Fcore-lro").

Reviewer 2: I would expect the samples to show the common ways to use core-lro. The link to the search results doesnt seem useful, one can always run search on their own.

Author: Sounds good! I can do that.
